src/server/callback/listener/helpers/handle-scan-results-callback.js

```diff
@@ -0,0 +1,46 @@
+import { fetcher } from '~/src/server/common/helpers/fetcher'
+import { toScanResultResponse } from '~/src/server/common/helpers/scan-result-response'
+import { deleteSqsMessage } from '~/src/server/scan/listener/helper/delete-sqs-message'
+import {
+  isAcknowledged,
+  uploadStatus
+} from '~/src/server/common/helpers/upload-status'
+
+async function handleScanResultsCallback(message, callbackQueueUrl, server) {
+  try {
+    const receiptHandle = message.ReceiptHandle
+    const payload = JSON.parse(message.Body)
+    const uploadDetails = await server.redis.findUploadWithFiles(
+      payload.uploadId
+    )
+    if (!uploadDetails) {
+      server.logger.error(
+        `Upload Id ${uploadDetails} not found. Deleting SQS message`
+      )
+      await deleteSqsMessage(server.sqs, callbackQueueUrl, receiptHandle)
+    } else if (!isAcknowledged(uploadDetails.uploadStatus)) {
+      const scanResult = toScanResultResponse(payload.uploadId, uploadDetails)
+      const url = uploadDetails.scanResultCallbackUrl
+      server.logger.error(`Failed to ${JSON.stringify(payload)}`)
```

Comment:
As an FYI - worth noting you can just pass in the payload like this - no need to `stringify`

`server.logger.error(payload, 'Failed')`

> not sure if the failed message makes sense here